app/models/opportunity.rb

```diff
@@ -1,5 +1,14 @@
 class Opportunity < ActiveRecord::Base
+  include PgSearch
+  pg_search_scope :search_by_city,
+    against: [:city]
+  pg_search_scope :search_by_cause,
+    associated_against: {
+    cause: [:cause]
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.